src/components/HistoryTable/historyTableContextReducer.ts

```diff
@@ -141,32 +156,50 @@ export const reducer = (state: HistoryTableReducerState, action: Action) => {
         ...state,
         columnLimit: action.limit,
       };
-    case "onChangeTableWidth": {
-      const nextColumnLimit = calcColumnLimitFromWidth(action.width);
-      if (nextColumnLimit === state.columnLimit) {
-        return state;
-      }
-      return {
-        ...state,
-        visibleColumns: state.columns.slice(0, nextColumnLimit),
-        columnLimit: nextColumnLimit,
-        currentPage: 0,
-        pageCount: Math.ceil(state.columns.length / nextColumnLimit),
-      };
-    }
     case "setHistoryTableFilters":
       return {
         ...state,
         historyTableFilters: action.filters,
       };
-    case "setSelectedCommit":
+    case "setSelectedCommit": {
+      let rowIndex = null;
+      let loaded = false;
+      const updatedProcessedCommits = state.processedCommits;
+      // If we already loaded the commit we need to search for it in the processed commits
+      // to set its selected state to true. This is in the case where we have the commit in the cache.
+      const hasMatchingCommit = state.commitCache.has(action.order);
+      if (hasMatchingCommit) {
+        rowIndex = state.processedCommits.findIndex((commit) =>
+          commitOrderToRowIndex(action.order, commit)
+        );
+        updatedProcessedCommits[rowIndex].selected = true;
```

Comment:
Do we need to do this on `L175`? I was fiddling around and printing the before and after values for `state.processedCommits` when this condition ran, and it seemed like the `selected` field was already marked `true` for the commit by default (which might make sense based on how `processCommits` is written)

Reply:
The bug was caused by `ingestCommits` and `setSelectedCommit` racing. Since `ingestCommits` would run immediately before `setSelectedCommit`  had a chance to run for commits that already existed in the cache. So if the commits are processed before we call set `setSelectedCommit` `processCommits` would have never set the selected field. Its not necessary for new commits but is for cached commits.